I suspect that we actually need to fix the code that is causing the compile warnings.

For those interested in resolving this the errors can be seen by opening the [Checks tab](https://github.com/ArduPilot/ardupilot/pull/28897/checks) just above the PR description, then expand QURT Build and then further expand the "Waf Build" above "'copter' finished successfully".  Below are at least some of the failures

```
[77](https://github.com/ArduPilot/ardupilot/actions/runs/29965103821/job/89074763976?pr=28897#step:3:179)
  ../../libraries/AC_PrecLand/AC_PrecLand_StateMachine.cpp:219:13: warning: absolute value function 'fabsf' given an argument of type 'double' but has parameter of type 'float' which may cause truncation of value [-Wabsolute-value]
          if (fabsf(pos.z - retry_pos_m.z) < MAX_POS_ERROR_M) {
              ^
  ../../libraries/AC_PrecLand/AC_PrecLand_StateMachine.cpp:219:13: note: use function 'std::abs' instead
  [ 55/965] Compiling libraries/AP_AHRS/AP_AHRS_NavEKF2.cpp
          if (fabsf(pos.z - retry_pos_m.z) < MAX_POS_ERROR_M) {
              ^~~~~
              std::abs
  1 warning generated.
```

```
  In file included from ../../libraries/AP_GPS/AP_GPS_UBLOX_CFGv2.cpp:21:
  ../../libraries/AP_GPS/AP_GPS_UBLOX.h:748:16: warning: static_assert with no message is a C++17 extension [-Wc++17-extensions]
      UBLOX_GNSS(UBLOX_GNSS_TYPE_ASSERT)
```

Plane's warnings are
```
[4](https://github.com/ArduPilot/ardupilot/actions/runs/29965103821/job/89074763976?pr=28897#step:3:1167)
  In file included from ../../libraries/APM_Control/AP_FW_Controller.cpp:20:
  ../../libraries/APM_Control/AP_FW_Controller.h:128:14: warning: private field 'last_run_ticks' is not used [-Wunused-private-field]
      uint32_t last_run_ticks;
               ^
  1 warning generated.
```

--- a/libraries/AP_Math/ftype.h
+++ b/libraries/AP_Math/ftype.h
@@ -28,7 +28,6 @@ typedef double ftype;
 #define ceilF(x) ceil(x)
 #define fminF(x,y) fmin(x,y)
 #define fmodF(x,y) fmod(x,y)
-#define fabsF(x) fabs(x)
 #define toftype todouble
 #else
 typedef float ftype;
@@ -47,7 +46,6 @@ typedef float ftype;
 #define ceilF(x) ceilf(x)
 #define fminF(x,y) fminf(x,y)
 #define fmodF(x,y) fmodf(x,y)
-#define fabsF(x) fabsf(x)
 #define toftype tofloat
 #endif
 

